### PR TITLE
[MCC-140730] Feature/delete

### PR DESCRIPTION
**Pull request by csavage-mdsol**
merge after #13 

PR Adds a delete spec, cleans up the delete to return a true no body 204.  Refactors all that .hale_json stuff to use a helper hale_url_for.

```
Finished in 37.88 seconds (files took 0.51877 seconds to load)
14 examples, 0 failures, 2 pending

Randomized with seed 43121
```

### CODE REVIEW

File: lib/moya/app/controllers/drds_controller.rb [hunk 1]
```diff
@@ -41,6 +40,28 @@ def create
     respond_with(@drd, options)
   end
 
+  def activate
+    @drd = get_drd
```
Comment (yzhangmedidata) [inline lib/moya/app/controllers/drds_controller.rb:44]: Can <code>@drd</code> be placed into a <code>before_filter</code> with the <code>get_drd</code> method

File: spec/integration/test_service_spec.rb [hunk 2]
```diff
@@ -28,28 +16,33 @@
...
+          { drd: { name: 'Pike',
+            status: 'activated',
+            kind: 'Roll-e',
+            leviathan_uuid: 'd34c78bd-583c-4eff-a66c-cd9b047417b4',
+            leviathan_url: 'http://example.org/leviathan/d34c78bd-583c-4eff-a66c-cd9b047417b4'
+          }
+        }
+      }
 
       it 'responds appropriately to root' do
```
Comment (yzhangmedidata) [inline spec/integration/test_service_spec.rb:31]: Should these test the response body also

File: spec/integration/test_service_spec.rb [hunk 3]
```diff
@@ -91,13 +79,74 @@
...
-      xit 'responds to an activate call' do
+      it 'responds idempotently to an activate call' do
+        # Create activated drd
+        req_body = { drd: {name: 'deactivated drd', status: 'deactivated', kind: 'Roll-e'},
+                     conditions: ['can_do_anything']
+                   }
+        response = post(create_url, req_body)
+
+        # Get the activate URL
+        drd = parse_hale(response.body)
```
Comment (yzhangmedidata) [inline spec/integration/test_service_spec.rb:108]: Should this be placed in a <code>before</code> block

File: spec/integration/test_service_spec.rb (hunk 2 above)
Comment (csavage-mdsol) [inline spec/integration/test_service_spec.rb:31]: Root is served by crichton middleware, I think the body here should be tested along with that middleware

File: spec/integration/test_service_spec.rb (hunk 3 above)
Comment (csavage-mdsol) [inline spec/integration/test_service_spec.rb:108]: If I could use nested contexts, before blocks would make more sense in a lot of these tests, but I can't without spinning the service up and down a kajillion times.

File: lib/moya/app/controllers/drds_controller.rb (hunk 1 above)
Comment (csavage-mdsol) [inline lib/moya/app/controllers/drds_controller.rb:44]: addressed in #15

**Pull request merged**